labgrid/util/helper.py

```diff
@@ -94,12 +94,15 @@ def log_callback(message):
         """Logs process output message along with its pid."""
         import logging
         logger = logging.getLogger("Process")
-        logger.info(message.decode(encoding="utf-8", errors="replace"))
+        message = message.decode(encoding="utf-8", errors="replace").strip("\n")
+        if message:
+            logger.info(message)
 
     @staticmethod
     def print_callback(message):
         """Prints process output message."""
-        print(message.decode(encoding="utf-8", errors="replace"))
+        message = message.decode(encoding="utf-8", errors="replace")
+        print("\r{}".format(message), end='')
```

Comment:
Won't this hide other output with doesn't end in a `\n`?

Reply:
Correct, I've pushed a fixup that appends `\n` to `buf` in case it's not already there.